src/main/deltachat/backup.js

```diff
@@ -1,9 +1,29 @@
 const C = require('deltachat-node/constants')
+const DeltaChat = require('deltachat-node')
 
 function backupExport (dir) {
   this._dc.importExport(C.DC_IMEX_EXPORT_BACKUP, dir)
 }
 
+function backupImport (file) {
+  let dc = new DeltaChat()
+  dc.open('/tmp/dc-backup', () => {
+    dc.importExport(C.DC_IMEX_IMPORT_BACKUP, file)
+  })
+  dc.on('ALL', (event, ...args) => {
+    console.log('backupImport:', event, ...args)
+  })
+  dc.on('2051', (progress) => {
+    this.emit('DC_EVENT_IMEX_PROGRESS', progress)
+    if(progress == '1000') {
+      let addr = dc.getConfig('addr')
+      this.emit('DD_EVENT_BACKUP_IMPORTED', addr)
```

Comment:
This is the tricky part, we need to spawn a new and empty dc context, start importing the backup in there, than figure out what the email address is and than move the folder to the right folder. If something goes wrong we also need to remove that temporary folder. @r10s is there maybe a way that we can get a method to extract the address out of a backup, without the need of a context? This way we could create the context already at the correct path which would make things easier.

Reply:
Also in my tests yesterday somehow the DC_EVEN_IMEX_PROGRESS event didn't get triggered correctly. Need to investigate this further.